### PR TITLE
storage: cleaning and adding mariaDB support

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -4,8 +4,8 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
 
 git_repository(
     name = "rules_iota",
-    commit = "896627184c7472b0065724a5807c12692e9d997c",
-    remote = "https://github.com/iotaledger/rules_iota.git",
+    commit = "f5f8933c683732367f5f8c5e033af07b50b29a4c",
+    remote = "https://github.com/thibault-martinez/rules_iota.git",
```
Comment (tsvisabo) [inline WORKSPACE]: Reminder to set remote back to it's original url

File: common/storage/tests/test_storage.c [hunk 2]
```diff
@@ -0,0 +1,440 @@
+/*
```
Comment (tsvisabo) [inline common/storage/tests/test_storage.c:1]: reminder to uncomment

File: common/storage/BUILD [hunk 3]
```diff
@@ -25,3 +39,31 @@ cc_library(
...
+genrule(
+    name = "tangle_db",
+    srcs =
+        select({
+            ":sqlite3": ["//common/storage/sql/sqlite3:tangle-schema"],
+            ":mariadb": ["//common/storage/sql/mariadb:tangle-schema"],
+            "//conditions:default": ["//common/storage/sql/sqlite3:tangle-schema"],
+        }),
+    outs = ["tangle.db"],
+    cmd = "$(location @sqlite3//:shell) $@ < $<",
```
Comment (tsvisabo) [inline common/storage/BUILD]: but that would mean that for either `sqlite3` or `mariadb` we will have a `tangle.db` generated by the `sqlite3` tool

File: common/storage/BUILD [hunk 4]
```diff
@@ -25,3 +39,31 @@ cc_library(
...
+genrule(
+    name = "spent_addresses_db",
+    srcs =
+        select({
+            ":sqlite3": ["//common/storage/sql/sqlite3:spent-addresses-schema"],
+            ":mariadb": ["//common/storage/sql/mariadb:spent-addresses-schema"],
+            "//conditions:default": ["//common/storage/sql/sqlite3:spent-addresses-schema"],
+        }),
+    outs = ["spent-addresses.db"],
+    cmd = "$(location @sqlite3//:shell) $@ < $<",
```
Comment (tsvisabo) [inline common/storage/BUILD]: ditto

File: ciri/consensus/spent_addresses/BUILD [hunk 5]
```diff
@@ -5,7 +5,7 @@ cc_library(
     visibility = ["//visibility:public"],
     deps = [
         "//common:errors",
-        "//common/storage/sql/sqlite3:sqlite3_storage",
+        "//common/storage/sql/sqlite3:storage_sqlite3",
```
Comment (tsvisabo) [inline ciri/consensus/spent_addresses/BUILD]: shouldn't we select here based on `config_settings`?

File: ciri/consensus/ledger_validator/tests/BUILD [hunk 6]
```diff
@@ -20,18 +20,10 @@ cc_test(
         "//ciri/consensus/test_utils",
         "//ciri/consensus/transaction_solidifier",
         "//common/helpers:digest",
-        "//common/storage/sql/sqlite3:sqlite3_storage",
-        "//common/storage/tests/helpers",
+        "//common/storage/sql/sqlite3:storage_sqlite3",
```
Comment (tsvisabo) [inline ciri/consensus/ledger_validator/tests/BUILD]: shouldn't we select here based on `config_settings` ?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: ciri/consensus/spent_addresses/BUILD (hunk 5 above)
Comment (thibault-martinez) [reply ciri/consensus/spent_addresses/BUILD]: Definitely

File: ciri/consensus/ledger_validator/tests/BUILD (hunk 6 above)
Comment (thibault-martinez) [reply ciri/consensus/ledger_validator/tests/BUILD]: Definitely

File: common/storage/BUILD (hunk 3 above)
Comment (thibault-martinez) [reply common/storage/BUILD]: Will fix

File: common/storage/BUILD (hunk 4 above)
Comment (thibault-martinez) [reply common/storage/BUILD]: Will fix

File: common/storage/tests/test_storage.c (hunk 2 above)
Comment (thibault-martinez) [reply common/storage/tests/test_storage.c:1]: Uncommenting them will be the goal of the next PR

File: WORKSPACE (hunk 1 above)
Comment (tsvisabo) [reply WORKSPACE]: ?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: WORKSPACE (hunk 1 above)
Comment (thibault-martinez) [reply WORKSPACE]: Needed the rules_iota PR to be merged first, will update now.

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**